AllReadyApp/Web-App/AllReady/Features/Notifications/VolunteerAssignedToItineraryHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Threading.Tasks;
+using AllReady.Models;
+using MediatR;
+using Microsoft.Data.Entity;
+
+namespace AllReady.Features.Notifications
+{
+    public class VolunteerAssignedToItineraryHandler : IAsyncNotificationHandler<VolunteerAssignedToItinerary>
+    {
+        private readonly AllReadyContext _context;
+        private readonly IMediator _mediator;
+
+        public VolunteerAssignedToItineraryHandler(AllReadyContext context, IMediator mediator)
+        {
+            _context = context;
+            _mediator = mediator;
+        }
+
+        public async Task Handle(VolunteerAssignedToItinerary notification)
+        {
+            var taskSignup = await _context.TaskSignups
+                .AsNoTracking()
+                .Include(x => x.User)
+                .SingleAsync(ts => ts.Id == notification.TaskSignupId)
+                .ConfigureAwait(false);
+
+            var emailAddress = !string.IsNullOrWhiteSpace(taskSignup.PreferredEmail) ? taskSignup.PreferredEmail : taskSignup.User?.Email;
+            var sms = !string.IsNullOrWhiteSpace(taskSignup.PreferredPhoneNumber) ? taskSignup.PreferredPhoneNumber : taskSignup.User?.PhoneNumber;
+
+            if (string.IsNullOrWhiteSpace(emailAddress) && string.IsNullOrWhiteSpace(sms)) return;
```

Comment:
@SeanFeldman, yes it would. As a matter of fact, after this soft release to June to the Red Cross, there is an item on the backlog that would encapsulate this logic for all messages, as well as let users set their communication preferences (SMS/emai, etc...) so we're not shuttling around messages that expose all communication types as properties and put the work on the consumer to look up and populate what needs to be sent.